### PR TITLE
Allow specifying java binary path

### PR DESCRIPTION
**Pull request by puremourning**
Validate and print a warning if the java detected on install is not java 11

### CODE REVIEW

File: ycmd/tests/java/server_management_test.py [hunk 1]
```diff
@@ -564,3 +565,17 @@ def ServerManagement_StartServer_Fails_test( app ):
                      has_entry( 'is_running', False )
                    ) )
                  ) )
+
+
+@IsolatedYcmdWithoutJava( { 'java_binary_path': '/this/path/does/not/exist' } )
+def ServerManagement_NoJavaPath_test( app ):
```
Comment (bstaletic) [inline ycmd/tests/java/server_management_test.py]: For these tests, we usually call `GetCompleter( user_options )` directly and assert that we got `None`.

For example:

```python
@patch( 'ycmd.completers.rust.rust_completer.RA_EXECUTABLE', 'does_not_exist' )
def GetCompleter_RANotFound_test( *args ):
  assert_that( not GetCompleter( user_options_store.GetAll() ) )
```

Also, these tests are located in `<filetype>_completer_test.py`.

Comment (puremourning) [reply ycmd/tests/java/server_management_test.py]: > For these tests, we usually call GetCompleter( user_options ) directly and assert that we got None.

Yeah I noticed that, but I prefer this way as it's less intrusive. 

> Also, these tests are located in <filetype>_completer_test.py

Fair, I'll move it.

**Comment by puremourning** [comment]
updated; hopefully now with actually working tests.

**Comment by bstaletic** [comment]
Looks good. I'll just wait for the tests to finish.

**Pull request merged**